core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -271,6 +276,26 @@ private void add(PendingDeleteFile file) {
     }
   }
 
+  protected void validateNewDeleteFile(DeleteFile file) {
+    Preconditions.checkNotNull(file, "Invalid delete file: null");
+    Preconditions.checkArgument(formatVersion() >= 2, "Deletes are supported in V2 and above");
+    if (file.content() == FileContent.POSITION_DELETES) {
+      Preconditions.checkArgument(
+          formatVersion() != 2 || !ContentFileUtil.isDV(file),
+          "Must not use DVs for position deletes in V2: %s",
+          ContentFileUtil.dvDesc(file));
+      Preconditions.checkArgument(
+          formatVersion() == 2 || ContentFileUtil.isDV(file),
```

Comment:
Same question here, why not just checking if it's DV: we know that if it's DV it's format version 3+ right ?

Reply:
This method validates input from the user and ensures it is not possible to add a DV to a V2 table.